packages/@aws-cdk/aws-appmesh/lib/private/utils.ts

```diff
@@ -92,3 +95,40 @@ export function renderMeshOwner(resourceAccount: string, meshAccount: string) :
     ? meshAccount
     : undefined;
 }
+
+/**
+ * This is the helper method to check if properties for passed match are undefined.
+ * This returns true if all properties for an object are undefined.
+ */
+export function areMatchPropertiesUndefined(match: HttpRouteMatch | GrpcRouteMatch | GrpcGatewayRouteMatch | undefined): boolean {
+  let isEmpty = true;
+  for (const property in match) {
+    if (property) {
+      isEmpty = false;
+      break;
+    }
+  }
+
+  return isEmpty;
+}
+
+/**
+ * This is the helper method to validate the length of Metadata array when it is specified.
+ */
+export function validateMetadata(metadata?: HeaderMatch[]) {
+  const MIN_LENGTH = 1;
+  const MAX_LENGTH = 10;
+
+  if (metadata && (metadata.length < MIN_LENGTH || metadata.length > MAX_LENGTH)) {
+    throw new Error(`Metadata must be between ${MIN_LENGTH} and ${MAX_LENGTH}`);
```

Comment:
Please include how many metadata were received in the error message.

Reply:
Addressed